components/ide/jetbrains/launcher/main.go

```diff
@@ -488,12 +490,25 @@ func resolveUserEnvs(launchCtx *LaunchContext) (userEnvs []string, err error) {
 	if err != nil {
 		return
 	}
-	envCmd := exec.Command(shell, []string{"-ilc", "/ide-desktop/jb-launcher env " + mark.String()}...)
+
+	self, err := os.Executable()
+	if err != nil {
+		return
+	}
+	envCmd := exec.Command(shell, []string{"-i", "-l", "-c", strings.Join([]string{self, "env", mark.String()}, " ")}...)
+	envCmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 	envCmd.Stderr = os.Stderr
+	envCmd.WaitDelay = 3 * time.Second
+
 	output, err := envCmd.Output()
-	if err != nil {
+	if errors.Is(err, exec.ErrWaitDelay) {
+		// For some reason the command doesn't close it's I/O pipes but it already run successfully
+		// so just ignore this error
+		log.Warn("WaitDelay expired before envCmd I/O completed")
```

Comment:
This is the actual fix

Reply:
maybe we should do it an error, it is rather a workaround, if it happens tools configuration will be likely bogus, like wrong PATH variable, JAVA_HOME, etc.

Reply:
actually it is not necessary, it will be already logged as part of `failed to resolve user env vars` below. We will double log it.

Reply:
It won't be logged because it will be overwritten later if a parsing error happens, I added the log so we can keep track in gcp how often this edge case happens